src/main/java/com/shapesecurity/functional/data/HashTable.java

```diff
@@ -516,6 +538,58 @@ private Fork(@Nonnull Hasher<K> hasher, @Nonnull HashTable<K, V>[] children, int
             return init;
         }
 
+        public Iterator<Pair<K, V>> iterator() {
+            return new Iterator<Pair<K, V>>() {
+                @SuppressWarnings("unchecked")
+                private final HashTable<K, V>[] stack = new HashTable[Fork.this.length];
+                private Maybe<Iterator<Pair<K, V>>> currentIterator = Maybe.empty();
```

Comment:
Because this is a small piece of performance-senstive, entirely internal code, it's OK to `null`s instead of `Maybe`s here.